mne/decoding/gsoc.py

```
@@ -0,0 +1,216 @@
+            labels of data.
+
+        Returns
+        -------
+        X : ndarray, shape(n_epochs, n_components * event_types * n_times)
+            spatially filtered signals.
+        """
+        self.fit(X, y)
+        return self.transform(X)
+
+    def inverse_transform(self, X, event_id=None, include=None, exclude=None):
+        """Remove selected components from the signal.
+
+        Given the unmixing matrix, transform data,
+        zero out components, and inverse transform the data.
+        This procedure will reconstruct M/EEG signals from which
+        the dynamics described by the excluded components is subtracted.
+
+        Parameters
+        ----------
+        X : np.ndarray, shape(n_epochs, n_components * event_types * n_times)
+            The signal data to undergo inverse transform.
+        event_id : dict | list of str | None (default None)
+            The kind of event to apply. if None, a dict of inst will be return
+            one for each type of event xdawn has been fitted.
+        include : array_like of int | None (default None)
+            The indices referring to columns in the ummixing matrix. The
+            components to be kept. If None, the first n_components (as defined
+            in the Xdawn constructor) will be kept.
+        exclude : array_like of int | None (default None)
+            The indices referring to columns in the ummixing matrix. The
+            components to be zeroed out. If None, all the components except the
+            first n_components will be exclude.
+
+        Returns
+        -------
+        out : dict of instance
+            A dict of instance (from the same type as inst input) for each
+            event type in event_id.
+        """
+
+        if not isinstance(X, np.ndarray):
+            raise ValueError("Given data should be numpy array, got "
+                             "%s instead" % type(X))
+
+        data = np.hstack(X.reshape(X.shape[0], self.n_chan, X.shape[1] /
```

No, it should only be int, else should throw an error.